contracts/TokenBridgeBase.sol

```diff
@@ -1,11 +1,12 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.17;
 
-import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 import {NonblockingLzApp} from "@layerzerolabs/solidity-examples/contracts/lzApp/NonblockingLzApp.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 /// @dev An abstract contract containing a common functionality used by OriginalTokenBridge and WrappedTokenBridge
-abstract contract TokenBridgeBase is NonblockingLzApp, ReentrancyGuard {
+abstract contract TokenBridgeBase is NonblockingLzApp, ReentrancyGuardUpgradeable, UUPSUpgradeable {
```

Comment:
We can create an upgradable version for `NonblockingLzApp`  (same as ReentrancyGuardUpgradeable)

Reply:
by forking?